llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -1227,6 +1261,30 @@ AArch64LoadStoreOpt::mergePairedInsns(MachineBasicBlock::iterator I,
             .addImm(0)
             .addImm(31);
     (void)MIBSXTW;
+
+    if (I->peekDebugInstrNum()) {
+      auto InstrNum = I->peekDebugInstrNum();
+      // If I is the instruction which got sign extended, restore the debug
+      // instruction number from I to the SBFMXri instruction
```

Comment:
I think this comment might benefit from an example (like in the PR description) to illustrate each case. This would make it obvious why two instructions get updated in the else branch, and only one in the then branch.